rmw_connextdds_common/include/rmw_connextdds/rmw_waitset_std.hpp

```diff
@@ -333,8 +335,38 @@ class RMW_Connext_StatusCondition : public RMW_Connext_Condition
   virtual bool
   has_status(const rmw_event_type_t event_type) = 0;
 
+  void
+  on_inconsistent_topic(const struct DDS_InconsistentTopicStatus * status);
+
+  void update_status_inconsistent_topic(const struct DDS_InconsistentTopicStatus * status);
+
+  inline rmw_ret_t
+  get_inconsistent_topic_status(
+    rmw_inconsistent_topic_status_t * const status)
+  {
+    update_state(
+      [this, status]() {
+        this->triggered_inconsistent_topic = false;
+
+        status->total_count = this->status_inconsistent_topic.total_count;
+        status->total_count_change = this->status_inconsistent_topic.total_count_change;
+
+        this->status_inconsistent_topic.total_count_change = 0;
+
+        this->status_inconsistent_topic_last = this->status_inconsistent_topic;
+      }, false /* notify */);
+
+    return RMW_RET_OK;
+  }
+
 protected:
   DDS_StatusCondition * scond;
+
+  bool triggered_inconsistent_topic{false};
+
+  struct DDS_InconsistentTopicStatus status_inconsistent_topic;
+
+  struct DDS_InconsistentTopicStatus status_inconsistent_topic_last;
```

Comment:
Do we need to keep the last status? Why do we not just store the current status from RTI connextdds to RMW?

Reply:
And this is now removed in 8f7660e